nearlib/wallet-access-key.js

```diff
@@ -0,0 +1,128 @@
+/**
+ * Wallet based account and signer that uses external wallet through the iframe to signs transactions.
+ */
+
+const KeyPair = require('./signing/key_pair');
+const BrowserLocalStorageKeystore = require('./signing/browser_local_storage_key_store');
+const SimpleKeyStoreSigner = require('./signing/simple_key_store_signer');
+
+const LOGIN_WALLET_URL_SUFFIX = '/login_v2/';
+const LOCAL_STORAGE_KEY_SUFFIX = '_wallet_access_key';
+
+/**
+ * Access Key based signer that uses Wallet to authorize app on the account and receive the access key.
+ * @example
+ * // if importing WalletAccessKey directly
+ * const walletAccount = new WalletAccessKey(contractName, walletBaseUrl)
+ * // if importing in all of nearLib and calling from variable
+ * const walletAccount = new nearlib.WalletAccessKey(contractName, walletBaseUrl)
+ * // To access this signer globally
+ * window.walletAccount = new nearlib.WalletAccessKey(config.contractName, walletBaseUrl);
+ * // To provide custom signer where the keys would be stored
+ * window.walletAccount = new nearlib.WalletAccessKey(config.contractName, walletBaseUrl, customSigner);
+ */
+class WalletAccessKey {
+    constructor(appKeyPrefix, walletBaseUrl = 'https://wallet.nearprotocol.com', signer = null) {
+        this._walletBaseUrl = walletBaseUrl;
+        this._authDataKey = appKeyPrefix + LOCAL_STORAGE_KEY_SUFFIX;
+        this._signer = signer || (new SimpleKeyStoreSigner(new BrowserLocalStorageKeystore()));
+
+        this._authData = JSON.parse(window.localStorage.getItem(this._authDataKey) || '{}');
+
+        if (!this.isSignedIn()) {
+            this._tryInitFromUrl();
+        }
+    }
+
+    /**
+     * Returns true, if this WalletAccount is authorized with the wallet.
+     * @example
+     * walletAccount.isSignedIn();
+     */
+    isSignedIn() {
+        return !!this._authData.accountId;
+    }
+
+    /**
+     * Returns authorized Account ID.
+     * @example
+     * walletAccount.getAccountId();
+     */
+    getAccountId() {
+        return this._authData.accountId || '';
+    }
+
+    /**
+     * Redirects current page to the wallet authentication page.
+     * @param {string} contract_id contract ID of the application
+     * @param {string} title name of the application
+     * @param {string} success_url url to redirect on success
+     * @param {string} failure_url url to redirect on failure
+     * @example
```

Comment:
let's not provide examples / comments which don't add extra information over method signature